test/e2e/registration_service_test.go

```diff
@@ -262,6 +262,29 @@ func (s *registrationServiceTestSuite) TestSignupFails() {
 		// Call signup endpoint with a valid token.
 		s.assertGetSignupReturnsNotFound(token1)
 	})
+
+	s.Run("get signup for crtadmin fails", func() {
+		// Get valid generated token for e2e tests. IAT claim is overridden
+		// to avoid token used before issued error. Username claim is also
+		// overridden to trigger error and ensure that usersignup is not created.
+		identity := authsupport.NewIdentity()
+		emailValue := uuid.NewV4().String() + "@acme.com"
+		emailClaim := authsupport.WithEmailClaim(emailValue)
+		usernameClaim := authsupport.WithPreferredUsernameClaim("test-crtadmin")
+		token, err := authsupport.GenerateSignedE2ETestToken(*identity, emailClaim, usernameClaim)
+		require.NoError(s.T(), err)
+
+		// Call signup endpoint with a valid token to initiate a signup process
+		response := invokeEndpoint(s.T(), "POST", s.route+"/api/v1/signup", token, "", http.StatusInternalServerError)
+		require.Equal(s.T(), "failed to create usersignup for test-crtadmin:cannot create usersignup for crtadmin", response["message"])
+		require.Equal(s.T(), "error creating UserSignup resource", response["details"])
+		require.Equal(s.T(), float64(500), response["code"])
```

Comment:
shouldn't it return `403` as `Forbidden`?

Reply:
Yeah.. so in reg-service code, i do return 403 forbidden, however, the handler just does a catch all and returns internal server error at the end: https://github.com/codeready-toolchain/registration-service/blob/master/pkg/controller/signup.go#L39-L43.. @alexeykazakov should this be changed?

Reply:
@tinakurian I would suggest that you also fix the return code to 403 in https://github.com/codeready-toolchain/registration-service/pull/170 for such a kind of username, indeed.

Reply:
I think that the error code should be changed/fixed to reflect the actual state/reason for the failure.
But feel free to do it as separate task/PR. @tinakurian Please create a jira story for that - as part of it go thought all the cases where we return error to check if there is any other use-case to be fixed.

Reply:
@xcoulon thanks for the review! yeah.. it needs to be fixed here though: https://github.com/codeready-toolchain/registration-service/blob/master/pkg/controller/signup.go#L39-L43.. the reg-service PR i created does return 403. It just isn't reflected in signup.go because there is a catch all which returns 500

Reply:
@MatousJobanek sounds good, i think i saw at least one other place where a 400s error is converted into a 500.. will do this in a separate PR and create a new Jira for it. also, thanks for the review!

Reply:
Jira for addressing the http error code catch-all: https://issues.redhat.com/browse/CRT-966